OpenRA.Mods.Common/ActorInitializer.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common
+{
+	public class FacingInit : IActorInit<int>
+	{
+		[FieldFromYamlKey]
+		readonly int value = 128;
+		public FacingInit() { }
+		public FacingInit(int init) { value = init; }
+		public virtual int Value(World world) { return value; }
+	}
+
+	public class DynamicFacingInit : FacingInit
```

Comment:
I mentioned this earlier, but it may have been lost in the scrum of other comments: can we please make this not subclass `FacingInit`?  My suggestions for the `GetFacing` and `GetOrientation` methods were specifically to enable this separation :smile:

The issue is that ActorInits are supposed to be immutable, and while this technically doesn't change that it means that in practice you can query something that you *think* is an immutable `FacingInit`, but is actually an immutable `DynamicFacingInit` that gives you different facing values each time you query it – violating expectations for a regular `FacingInit`.